packages/vite/src/node/plugins/css.ts

```diff
@@ -885,38 +926,73 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
         })
       }
 
-      function extractCss() {
+      async function extractCss() {
         let css = ''
-        const collected = new Set<OutputChunk>()
+        let line = 0
+        const collected = new Set<OutputAsset | OutputChunk>()
+        const concatCssEndLines: Array<{ file: string; end: number }> = []
         const prelimaryNameToChunkMap = new Map(
           Object.values(bundle)
             .filter((chunk): chunk is OutputChunk => chunk.type === 'chunk')
             .map((chunk) => [chunk.preliminaryFileName, chunk]),
         )
 
-        function collect(fileName: string) {
+        function collect(fileName: string): string {
           const chunk = bundle[fileName]
-          if (!chunk || chunk.type !== 'chunk' || collected.has(chunk)) return
-          collected.add(chunk)
+          if (!chunk || chunk.type !== 'chunk') {
+            collected.add(chunk)
+            return ''
+          }
+          if (collected.has(chunk)) {
+            return ''
+          }
+
+          const css = chunkCSSMap.get(chunk.preliminaryFileName) ?? ''
+          const importedCSS = chunk.imports.reduce((css, file) => {
+            const imported = collect(file)
+            if (imported.length > 0) {
+              return css + '\n' + imported
+            }
+            return css
+          }, '')
 
-          chunk.imports.forEach(collect)
-          css += chunkCSSMap.get(chunk.preliminaryFileName) ?? ''
+          return importedCSS + '\n' + css
         }
 
-        for (const chunkName of chunkCSSMap.keys())
-          collect(prelimaryNameToChunkMap.get(chunkName)?.fileName ?? '')
+        for (const chunkName of chunkCSSMap.keys()) {
+          const filename = prelimaryNameToChunkMap.get(chunkName)?.fileName
+          if (!filename) {
+            continue
+          }
+
+          const cssCode = collect(filename)
+
+          if (cssCode) {
+            line += getLineCount(cssCode)
+            concatCssEndLines.push({ file: filename, end: line })
+
+            css += cssCode
+          }
+        }
 
-        return css
+        return await finalizeCss(
+          css,
+          false,
+          config,
+          undefined,
```

Comment:
```suggestion
          cssBundleName,
```
I think passing `cssBundleName` makes sense.